client/task_runner_test.go

```diff
@@ -1614,3 +1614,67 @@ func TestTaskRunner_Pre06ScriptCheck(t *testing.T) {
 	t.Run(run("0.5.6", "java", "tcp", false))
 	t.Run(run("0.5.6", "mock_driver", "tcp", false))
 }
+
+func TestTaskRunner_ShutdownDelay(t *testing.T) {
+	t.Parallel()
+
+	alloc := mock.Alloc()
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Driver = "mock_driver"
+	task.Config = map[string]interface{}{
+		"run_for": "1000s",
+	}
+
+	// No shutdown escape hatch for this delay, so don't set it too high
+	task.ShutdownDelay = 5 * time.Second
```

Comment:
If we're unable to check the Consul ops before the delay is up we'll get a false failure. Mind if I make it `2s` to give Travis plenty of time? Since this test runs in parallel with the rest of `client/` -- which takes ~50s total -- it should slow down the overall wallclock time.